roles/postgres_upgrade/tasks/main.yml

```diff
@@ -0,0 +1,57 @@
+---
+
+# This view for Datadog references system tables, which blows up if we try to upgrade the cluster
+# with it included. We can safely drop it here and re-add it later via the datadog role.
+- name: drop pg_stat_activity_dd view # noqa 301
+  command: "psql {{ db }} -c 'DROP VIEW IF EXISTS pg_stat_activity_dd CASCADE;'"
+  become: yes
+  become_user: postgres
+
+- name: install new version
+  include_tasks: install.yml
+
+- name: stop postgres
+  service:
+    name: postgresql
+    state: stopped
+  become: yes
+
+- name: check upgrade compatibility # noqa 301
+  command: >
+    /usr/lib/postgresql/{{ pg_new }}/bin/pg_upgrade --link --check
+    --old-options=-cconfig_file=/etc/postgresql/{{ pg_old }}/main/postgresql.conf
+    --new-options=-cconfig_file=/etc/postgresql/{{ pg_new }}/main/postgresql.conf
+  args:
+    chdir: "/var/lib/postgresql"
+  become: yes
+  become_user: postgres
+  environment:
+    PGDATAOLD: "/var/lib/postgresql/{{ pg_old }}/main"
+    PGDATANEW: "/var/lib/postgresql/{{ pg_new }}/main"
+    PGBINOLD: "/usr/lib/postgresql/{{ pg_old }}/bin"
+    PGBINNEW: "/usr/lib/postgresql/{{ pg_new }}/bin"
+  register: compatibility_check
+
+- name: perform upgrade
+  command: >
+    /usr/lib/postgresql/{{ pg_new }}/bin/pg_upgrade --link
+    --old-options=-cconfig_file=/etc/postgresql/{{ pg_old }}/main/postgresql.conf
+    --new-options=-cconfig_file=/etc/postgresql/{{ pg_new }}/main/postgresql.conf
+  args:
+    chdir: "/var/lib/postgresql"
+  become: yes
+  become_user: postgres
+  environment:
+    PGDATAOLD: "/var/lib/postgresql/{{ pg_old }}/main"
+    PGDATANEW: "/var/lib/postgresql/{{ pg_new }}/main"
+    PGBINOLD: "/usr/lib/postgresql/{{ pg_old }}/bin"
+    PGBINNEW: "/usr/lib/postgresql/{{ pg_new }}/bin"
+  when: compatibility_check is succeeded
+  register: upgrade_result
+
+- debug:
+    var: upgrade_result.stdout_lines
```

Comment:
I guess we want to keep this, right? we'll see `pg_upgrade`'s output, right?

Reply:
I was thinking of dropping this and just recommending we only run the playbook with the `-vv` option.